### DISCUSSION TITLE
Issue with file names.

### DISCUSSION THREAD

**Discussion by lessirey** [basujindal/stable-diffusion#26] [Ideas]
The Folder and file names try to include the entire prompt, which is good.. but on lengthy prompts the system breaks and gives an error that file cannot be found since (windows) cant create such a lengthy name. its fine on smaller prompts.

Ideally its good to include the prompt but truncated to specific length and end with seed value.
eg:
[beginning part of the prompt, but truncated]+[addtitional parameters]+[seed value].png

**Comment by GucciFlipFlops1917** [comment]
To add on, I had such an issue where the files were not saved due to the length of the prompt. It was worse as I adjusted the save file to include the prompt. I made some more adjustments, so my line numbers differ from the official script, but areas to adjust in the txt2img file would be the following: 

~Line 160:
`sample_path = os.path.join(outpath, "_".join(opt.prompt.split()))[:150]`

~Line 275-277:
`                   adjustprompt = prompt.replace(" ", "_")[:150]
                    Image.fromarray(x_sample.astype(np.uint8)).save(
                        os.path.join(sample_path, "seed_" + str(opt.seed) + "_" + f"{adjustprompt}{base_count:05}.png"))`

Where, the "[:150]" can be shortened in the instance of issues.

**Comment by Michoko92** [comment]
FYI, I think it is possible to remove the 260 characters limits on WIndows 10. See here : https://www.howtogeek.com/266621/how-to-make-windows-10-accept-file-paths-over-260-characters/
